pkg/capr/planner/config.go

```diff
@@ -356,10 +357,32 @@ func addToken(config map[string]interface{}, entry *planEntry, tokensSecret plan
 }
 
 func addAddresses(secrets corecontrollers.SecretCache, config map[string]interface{}, entry *planEntry) error {
+	// Helpers
+	firstNonEmpty := func(a, b string) string {
+		if a != "" {
+			return a
+		}
+		return b
+	}
+	appendNonEmptyUnique := func(m map[string]interface{}, key, val string) {
+		if val == "" {
+			return
+		}
+		cur := convert.ToStringSlice(m[key])
+		for _, existing := range cur {
+			if existing == val {
+				return
+			}
+		}
+		m[key] = append(cur, val)
+	}
```

Comment:
I would honestly do without the helpers. you can just as easily write:
```
nodeIP := internalIPAddress
if nodeIP == "" {
  nodeIP := ipAddress
}

if nodeIP != "" && !slices.Contains(convert.ToStringSlice(config["node-ip"]), nodeIP) {
  config["node-ip"] = append(convert.ToStringSlice(config["node-ip"]), nodeIP)
}
```

`convert.ToStringSlice` just does a type cast if the data was already a string (which it should be). 

If you really want to have a function, I would make a function like
```
func assignIPAddresses(config map[string]any, ip, fallback string)
```

Reply:
Good point! 
The logic has been updated to eliminate the helper functions.